pkg/azurefile/azurefile.go

```diff
@@ -1279,3 +1282,32 @@ func (d *Driver) getFileShareClientForSub(subscriptionID string) (fileshareclien
 	}
 	return d.cloud.ComputeClientFactory.GetFileShareClientForSub(subscriptionID)
 }
+
+func getNodeInfoFromLabels(ctx context.Context, nodeID string, kubeClient clientset.Interface) (string, string, error) {
+	if kubeClient == nil || kubeClient.CoreV1() == nil {
+		return "", "", fmt.Errorf("kubeClient is nil")
+	}
+
+	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeID, metav1.GetOptions{})
+	if err != nil {
+		return "", "", fmt.Errorf("get node(%s) failed with %v", nodeID, err)
+	}
+
+	if len(node.Labels) == 0 {
+		return "", "", fmt.Errorf("node(%s) label is empty", nodeID)
+	}
+	return node.Labels["kubernetes.azure.com/kata-mshv-vm-isolation"], node.Labels["katacontainers.io/kata-runtime"], nil
+}
+
+func isKataNode(ctx context.Context, nodeID string, kubeClient clientset.Interface) bool {
+	kataVMIsolationLabel, kataRuntimeLabel, err := getNodeInfoFromLabels(ctx, nodeID, kubeClient)
```

Comment:
if nodeID == "" {
  // driver controller
  return false
}

Reply:
thus in driver controller pod, it does not need to get node labels since nodeID is empty